pkg/jobrunaggregator/jobrunaggregatorlib/big_query.go

```diff
@@ -92,8 +95,36 @@ func (d dryRunInserter) Put(ctx context.Context, src interface{}) (err error) {
 	buf := &bytes.Buffer{}
 	fmt.Fprintf(buf, "BULK INSERT into %v\n", d.table)
 	for i := 0; i < srcVal.Len(); i++ {
-		s := srcVal.Index(i).Interface()
-		fmt.Fprintf(buf, "\tINSERT into %v: %v\n", d.table, s)
+
+		// Infer the struct type from the table name
+		switch d.table {
+		case jobrunaggregatorapi.TestRunTableName:
+			s := srcVal.Index(i).Interface().(*jobrunaggregatorapi.TestRunRow)
+			fmt.Fprintf(buf, "\tINSERT into %v: prowJob=%v, ts=%v, lts=%v, tcn=%v\n", d.table, s.ProwJob.ObjectMeta.Name,
+				s.TestSuites[0], len(s.TestSuites), s.TestCase.Name)
+			if len(s.TestSuites) > 1 {
+				fmt.Println("Found a testSuite list greater than one")
+			}
+
+		case jobrunaggregatorapi.LegacyJobRunTableName, jobrunaggregatorapi.DisruptionJobRunTableName:
+			s := srcVal.Index(i).Interface().(*jobrunaggregatorapi.JobRunRow)
+			fmt.Fprintf(buf, "\tINSERT into %v: name=%v, jobname=%v, status=%v\n", d.table, s.Name, s.JobName, s.Status)
+
+		case jobrunaggregatorapi.BackendDisruptionTableName:
+			s := srcVal.Index(i).Interface().(*jobrunaggregatorapi.BackendDisruptionRow)
+			fmt.Fprintf(buf, "\tINSERT into %v: BEName=%v, JobRunName=%v, DisSecs=%v\n", d.table, s.BackendName, s.JobRunName, s.DisruptionSeconds)
+
+		case jobrunaggregatorapi.JobsTableName:
+			s := srcVal.Index(i).Interface().(jobrunaggregatorapi.JobRow)
+			fmt.Fprintf(buf, "\tINSERT into %v: JobName=%v\n", d.table, s.JobName)
+
+		default:
+
+			// Today, NewDryRunInserter is created for the above types.  If there are more,
+			// abort so that we know to add another case block above.
+			fmt.Println("Unknown table in dry-run mode")
+			os.Exit(1)
```

Comment:
don't exit from library code.  Given the intent of this code, simply writing the error will do.